### PR TITLE
UniPhier Initial Support

### PR DESCRIPTION
**Pull request by masahir0y**

### CODE REVIEW

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by masahir0y** [comment]
Sorry, please let me fix one line.
I found unnecessary #include <debug.h>.
(I used INFO() function for debugging, but I forgot to remove the include).

I will push a fixed version soon.

**Comment by masahir0y** [comment]
The branch has been updated.

One minor fix-up:
  - Remove unneeded <debug.h> inclusion from uniphier_image_desc.c

**Comment by davidcunado-arm** [comment]
jenkins: please test this

**Comment by davidcunado-arm** [comment]
jenkins: test this please

File: plat/socionext/uniphier/uniphier_nand.c [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+	do {
+		status = mmio_read_32(nand->reg_base + DENALI_INTR_STATUS0);
+	} while (!(status & DENALI_INTR_INT_ACT));
+
+	bbm = uniphier_nand_host_read(nand, DENALI_MAP11_DATA);
+
+	is_bad = bbm != 0xff;
+
+	/* save the result for future re-use */
+	nand->bbt[block] = is_bad;
```
Comment (davidcunado-arm) [inline plat/socionext/uniphier/uniphier_nand.c]: @masahir0y 
Our static analysis tool picked-up an issue here - possible out of bounds write of bbt array. Would be worth having an assert that checks that block is within the bounds of the array.

Comment (masahir0y) [inline plat/socionext/uniphier/uniphier_nand.c]: @davidcunado-arm 

Thanks for catching this.  This is a bug that must be absolutely fixed.
If the PR had stayed in the integration branch, I would have been able to pushed a fix version, but it has gone to the master branch.   I will open a new pull request for the fixup.

**Pull request merged**